### PR TITLE
ledger refactoring: catchpointdump support of new schema

### PR DESCRIPTION
**Pull request by algorandskiy**
## Summary

1. Generalized all accounts and resources reading from DB via callbacks
2. Use this new reading code in catchpointdump

This also requires changes from #3229

## Test Plan

TODO

### CODE REVIEW

File: ledger/accountdb.go [hunk 1]
```diff
@@ -2627,85 +2794,39 @@ func (iterator *orderedAccountsIter) Next(ctx context.Context) (acct []accountAd
...
-			hash := accountHashBuilderV6(addr, &accountData, buf)
-			_, err = iterator.insertStmt.ExecContext(ctx, addrbuf, hash)
-			if err != nil {
-				iterator.Close(ctx)
 				return
 			}
+			return nil
+		}
 
-			resourcesEntriesCount := accountData.TotalAppParams + accountData.TotalAppLocalStates + accountData.TotalAssetParams + accountData.TotalAssets
```
Comment (algorandskiy) [inline ledger/accountdb.go:2663]: this block was moved to `processAllResources` func

File: ledger/accountdb.go [hunk 2]
```diff
@@ -2573,6 +2573,175 @@ func makeOrderedAccountsIter(tx *sql.Tx, accountCount int) *orderedAccountsIter
...
+				ad.Assets = make(map[basics.AssetIndex]basics.AssetHolding)
+			}
+			ad.Assets[basics.AssetIndex(cidx)] = resData.GetAssetHolding()
+			resCounter++
+		}
+
+		return nil
+	}
+
+	if baseAcct.TotalAppParams+baseAcct.TotalAppLocalStates+baseAcct.TotalAssetParams+baseAcct.TotalAssets == resCounter {
```
Comment (tsachiherman) [inline ledger/accountdb.go:2730]: I can't see where this variable is being populated.

Comment (algorandskiy) [reply ledger/accountdb.go:2730]: `baseCb` sets `baseAcct` variable, and `resCb` updates the `resCounter` counter

File: ledger/accountdb.go [hunk 3]
```diff
@@ -2573,6 +2573,175 @@ func makeOrderedAccountsIter(tx *sql.Tx, accountCount int) *orderedAccountsIter
...
+}
+
+// LoadAllFullAccounts loads all accounts from balancesTable and resourcesTable.
+// On every account full load it invokes acctCb callback to report progress and data.
+func LoadAllFullAccounts(
+	ctx context.Context, tx *sql.Tx,
+	balancesTable string, resourcesTable string,
+	acctCb func(basics.Address, basics.AccountData),
+) (count int, err error) {
+	baseRows, err := tx.QueryContext(ctx, fmt.Sprintf("SELECT rowid, address, data FROM %s ORDER BY rowid", balancesTable))
```
Comment (tsachiherman) [inline ledger/accountdb.go:2674]: maybe `defer baseRows.Close()` and `defer resRows.Close()` ?

File: ledger/accountdb.go (hunk 2 above)
Comment (tsachiherman) [reply ledger/accountdb.go:2730]: yes, but the `baseAcct` is called only via `processAllBaseAccountRecords`, and the above comparison is made prior to that.

Comment (algorandskiy) [reply ledger/accountdb.go:2733]: oh, right, good catch. fixed.

**Review by tsachiherman — APPROVED** [review]
looks good.

**Pull request merged**